fs.go

```diff
@@ -1158,10 +1204,22 @@ func (h *fsHandler) newFSFile(f *os.File, fileInfo os.FileInfo, compressed bool)
 		contentType = http.DetectContentType(data)
 	}
 
+	var buffer []byte
+	if !store {
+		buffer = nil
+	} else {
+		buffer = make([]byte, contentLength)
+		n, err := f.Read(buffer)
+		if err != nil || n != contentLength {
+			buffer = nil
```

Comment:
Probably better to return the error here?

Reply:
I don't want to break the file serving for reading problems. I can return error code in the final return sentence and maybe handle it out of the function. What do you think?